Panic from attempted access of array element in "by" grouping

Repro is with `zq` commit `0dba379a`.

Let's say I've got a `null` string in my input. If I try to access it like an array with `put`, I get a helpful error message.

```
$ echo '{"mynull": null}' | zq -f zson 'put a=mynull[0]' -
cannot index type "string" with key "int64: ()"
{
    mynull: null (string)
} (=0)
```

However, if I attempt a similar access in a `by` grouping of an aggregation, I get a panic.

```
$ echo '{"mynull": null}' | zq -f zson 'count() by a=mynull[0]' -
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x155d814]

goroutine 69 [running]:
github.com/brimsec/zq/zng/resolver.recordKey(0xc000a12400, 0x2, 0x2, 0x5, 0xc000717c10)
	/Users/phil/work/zq/zng/resolver/context.go:125 +0xb4
github.com/brimsec/zq/zng/resolver.(*Context).LookupTypeRecord(0xc00061e4c0, 0xc000a12400, 0x2, 0x2, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/zng/resolver/context.go:228 +0x1ac
github.com/brimsec/zq/proc/groupby.(*Aggregator).lookupRecordType(0xc0000e2120, 0xc00060ec80, 0x2, 0x2, 0xc000036100, 0x1, 0x8)
	/Users/phil/work/zq/proc/groupby/groupby.go:554 +0x114
github.com/brimsec/zq/proc/groupby.(*Aggregator).readTable(0xc0000e2120, 0xc000600001, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/proc/groupby/groupby.go:525 +0x49b
github.com/brimsec/zq/proc/groupby.(*Aggregator).Results(0xc0000e2120, 0x1, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/proc/groupby/groupby.go:359 +0xde
github.com/brimsec/zq/proc/groupby.(*Proc).run(0xc000a12300)
	/Users/phil/work/zq/proc/groupby/groupby.go:178 +0x354
created by github.com/brimsec/zq/proc/groupby.(*Proc).Pull.func1
	/Users/phil/work/zq/proc/groupby/groupby.go:158 +0x3e
```

--- a/expr/agg/ztests/groupby-missing.yaml
+++ b/expr/agg/ztests/groupby-missing.yaml
@@ -1,0 +1,9 @@
+zql: |
+  by a=a[0], b=b.c
+
+input: |
+   {a: "a", b: "b"}
+   {a: [0], b: {c: 1}}
+
+output: |
+  {a:0,b:1}

--- a/expr/agg/ztests/groupby-missing.yaml
+++ b/expr/agg/ztests/groupby-missing.yaml
@@ -3,6 +3,7 @@ zql: |
 
 input: |
    {a: "a", b: "b"}
+   {a: [], b: {c: 1}}
    {a: [0], b: {c: 1}}
 
 output: |

--- a/expr/eval.go
+++ b/expr/eval.go
@@ -616,7 +616,7 @@ func (i *Index) Eval(rec *zng.Record) (zng.Value, error) {
 	case *zng.TypeMap:
 		return indexMap(typ, container.Bytes, index)
 	default:
-		return zng.Value{}, fmt.Errorf("cannot index type \"%s\" with key \"%s\"", typ, index)
+		return zng.Value{}, zng.ErrMissing
 	}
 }
 

--- a/proc/groupby/groupby.go
+++ b/proc/groupby/groupby.go
@@ -271,8 +271,8 @@ func (a *Aggregator) Consume(r *zng.Record) error {
 			if errors.Is(err, zng.ErrMissing) {
 				// block this input type
 				a.block[id] = struct{}{}
-				return nil
 			}
+			return nil
 		}
 		if i == 0 && a.inputSortDir != 0 {
 			prim = a.updateMaxTableKey(zv)

